src/libstd/io/stdio.rs

```diff
@@ -872,6 +872,100 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Equivalent to the [`input`] method except that
+/// a prompt is printed before capturing input.
+///
+/// # Note
+///
+/// This function is not the recommended solution to get user input, it should
+/// only be used for simple actions. If you require more explicit control over
+/// getting user input, see the [`Stdin::read_line`] method.
+///
+/// # Panics
+///
+/// This function currently panics if it does not receive input,
+/// or when input is empty without newline
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(prompt)]
+/// use std::io::prompt;
+///
+/// fn main() {
+///     let user_input = prompt("Please enter some text: ");
+///
+///     println!("You typed: {}", user_input);
+/// }
+/// ```
+#[unstable(
+    feature = "prompt",
+    reason = "this function may be replaced with a \
+                     more general mechanism",
+    issue = "none"
+)]
+pub fn prompt(prompt: &str) -> String {
+    stdout().write(prompt.as_bytes()).unwrap();
+    input()
+}
+
+/// Constructs a new handle to the standard input of the current
+/// process, locks this handle and reads a line of input,
+/// returning a `String` containing the input. For automatic
+/// prompt handling, see the [`prompt`] method.
+///
+/// If you need more explicit control over
+/// locking, see the [`Stdin::lock`] and [`Stdout::lock`] methods.
+///
+/// # Note
+///
+/// This function is not the recommended solution to get user input, it should
+/// only be used for simple actions. If you require more explicit control over
+/// getting user input, see the [`Stdin::read_line`] method.
+///
+/// # Panics
+///
+/// This function currently panics if it does not receive input,
+/// or when input is empty without newline
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(input)]
+/// use std::io::input;
+///
+/// fn main() {
+///     print!("Please enter some text: ");
+///
+///     let user_input = input();
+///
+///     println!("You typed: {}", user_input);
+/// }
+/// ```
+#[unstable(
+    feature = "input",
+    reason = "this function may be replaced with a \
+                     more general mechanism",
+    issue = "none"
+)]
+pub fn input() -> String {
+    let mut input = String::new();
+    let _ = stdout().flush();
+    let len = stdin().read_line(&mut input).expect("Failed to read from stdin");
+    match len {
+        0 => panic!("invalid character!"),
+        _ => {
+            if input.as_bytes()[input.len() - 1] == b'\n' {
+                input.pop();
+            }
+            if input.as_bytes()[input.len() - 1] == b'\r' {
+                input.pop();
+            }
```

Comment:
This will panic with `attempt to subtract with overflow` if the input only consist of `\n`.

Reply:
Added cross-platform whitespace trimming